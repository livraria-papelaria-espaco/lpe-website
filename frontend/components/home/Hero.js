import { Typography, Container, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'relative',
    background: `linear-gradient(45deg, ${theme.palette.primary.dark} 20%, ${theme.palette.primary.light} 70%)`,
    color: theme.palette.primary.contrastText,
    marginTop: -theme.spacing(4),
    minHeight: '50vh',
    display: 'flex',
    flexDirection: 'column',
    overflow: 'hidden',
  },
  after: {
    position: 'absolute',
    borderRadius: '80% 80% 0 0 / 70% 70% 0 0',
    background: theme.palette.background.default,
    height: '20vh',
    width: '120%',
    bottom: '-15vh',
    left: 0,
    transform: 'translateX(-10%)',
  },
  container: {
    flexGrow: 1,
    paddingTop: theme.spacing(10),
    paddingBottom: theme.spacing(6),
  },
  containedButton: {
    backgroundColor: theme.palette.primary.contrastText,
  },
  button: {
    marginRight: theme.spacing(2),
  },
}));

const Hero = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Container fixed className={classes.container}>
        <Typography variant='h3' component='h1' gutterBottom>
          Livraria e Papelaria Espa√ßo
        </Typography>
        <div>
          <Button
            variant='contained'
            color='default'
            size='medium'
            className={`${classes.button} ${classes.containedButton}`}
          >
            Action 1
          </Button>
          <Button variant='outlined' color='inherit' className={classes.button}>
            Action 2
          </Button>
        </div>
      </Container>
      <div className={classes.after} />
    </div>
  );
};

export default Hero;
