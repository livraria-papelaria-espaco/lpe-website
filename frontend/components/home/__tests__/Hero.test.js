import { shallow } from 'enzyme';
import React from 'react';
import Hero from '../Hero';

describe('Hero static component (components/home/Hero)', () => {
  it('Should render correctly and match snapshot', () => {
    process.env.appbar = {
      desktopHeight: 96,
    };

    const wrapper = shallow(<Hero />);

    expect(wrapper).toMatchSnapshot();
  });
});
