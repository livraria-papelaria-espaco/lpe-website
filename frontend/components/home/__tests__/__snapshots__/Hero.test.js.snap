// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero static component (components/home/Hero) Should render correctly and match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-3"
    fixed={true}
  >
    <LogoSvg
      className="makeStyles-logo-4"
      viewBox="0 0 359.03 313.42"
      xmlns="http://www.w3.org/2000/svg"
    />
    <WithStyles(ForwardRef(Typography))
      component="h1"
      gutterBottom={true}
      variant="h3"
    >
      Livraria e Papelaria Espaço
    </WithStyles(ForwardRef(Typography))>
    <div>
      <Link
        href="/search"
        passHref={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-6 makeStyles-containedButton-5"
          color="default"
          size="medium"
          variant="contained"
        >
          Visitar loja
        </WithStyles(ForwardRef(Button))>
      </Link>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-6"
        color="inherit"
        component={[Function]}
        href="#about-us"
        variant="outlined"
      >
        Sobre nós
      </WithStyles(ForwardRef(Button))>
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
